### ISSUE TITLE
Training stops without error

### ISSUE THREAD

**Issue by Vichoko** [williamFalcon/pytorch-lightning#487]
**Describe the bug**
The training stops without user interaction at an arbitrary epoch. I don't know why it stops. 

**To Reproduce**
I use the template scheme proposed in the example code.
```

        logger = ptl.logging.TestTubeLogger(
            save_dir=MODELS_DATA_PATH / experiment_name,
            version=1  # An existing version with a saved checkpoint
        )
        trainer = ptl.Trainer(
            gpus=hyperparams.gpus,
            distributed_backend=hyperparams.distributed_backend,
            logger=logger,
            default_save_path=MODELS_DATA_PATH / experiment_name
        )
        trainer.fit(model)
```

**Expected behavior**
As I have not stated the epoch limit or stopping criteria explicitly, it should train indefinitely. 
If is a lightning feature it should say the stoppin criteria.

**Screenshots**
```
(aidio) [vichoko@pumba aidio]$ python train.py  --model waveNetTransformer --features_path ~/data/ --gpus 1
info: experiment_name is wavenet_transformer_pure_1
Epoch 11: 100%|██████████████████████████████████████████████████████████████████████████| 74/74 [16:04<00:00, 13.04s/batch, batch_nb=48, gpu=0, loss=1.337, train_loss=1.09, v_nb=1, val_acc=0.347, val_loss=1.34]
(aidio) [vichoko@pumba aidio]$
```

**Desktop (please complete the following information):**
 - OS Arch Linux
 - Version  0.5.3.2

**Comment by williamFalcon** [comment]
run it with coolmodel first. if it works fine then u probably have something in your lightningmodule that exits early

**Comment by Vichoko** [comment]
I have it have something to do to metrics not improving during several sequential epochs. Because the checkpoint saved is from epoch 3, and by epoch 11 metrics hasn't got any better.
Could it be this?

My module is very similar to this one, as you can see here:

```Python
class WavenetTramsformerClassifier(ptl.LightningModule):
    """
    Sample model to show how to define a template
    """

    def __init__(self, hparams, num_classes, train_dataset, eval_dataset, test_dataset):
        super(WavenetTramsformerClassifier, self).__init__()
        self.hparams = hparams
        self.loss = torch.nn.CrossEntropyLoss()
        self.train_dataset = train_dataset
        self.eval_dataset = eval_dataset
        self.test_dataset = test_dataset
        # build model
        self.model = WaveNetTransformerClassifier(num_classes)
        self.optimizer = torch.optim.Adam(self.model.parameters(), lr=hparams.learning_rate)

    # ---------------------
    # TRAINING
    # ---------------------
    def forward(self, x):
        """
        No special modification required for lightning, define as you normally would
        :param x:
        :return:
        """
        return self.model(x)

    def training_step(self, batch, batch_idx):
        """
        Lightning calls this inside the training loop
        :param batch:
        :return:
        """
        # forward pass
        x, y = batch['x'], batch['y']
        y_pred = self.forward(x)

        # calculate loss
        loss_val = self.loss(y_pred, y)

        tqdm_dict = {'train_loss': loss_val}
        output = OrderedDict({
            'loss': loss_val,
            'progress_bar': tqdm_dict,
            'log': tqdm_dict
        })

        # can also return just a scalar instead of a dict (return loss_val)
        return output

    def validation_step(self, batch, batch_idx):
        """
        Lightning calls this inside the validation loop
        :param batch:
        :return:
        """
        x, y = batch['x'], batch['y']
        y_pred = self.forward(x)

        # calculate loss
        loss_val = self.loss(y_pred, y)

        # acc
        labels_hat = torch.argmax(y_pred, dim=1)
        val_acc = torch.sum(y == labels_hat).item() / (len(y) * 1.0)
        val_acc = torch.tensor(val_acc)

        if self.on_gpu:
            val_acc = val_acc.cuda(loss_val.device.index)

        output = OrderedDict({
            'val_loss': loss_val,
            'val_acc': val_acc,
        })

        # can also return just a scalar instead of a dict (return loss_val)
        return output

    def validation_end(self, outputs):
        """
        Called at the end of validation to aggregate outputs
        :param outputs: list of individual outputs of each validation step
        :return:
        """
        # if returned a scalar from validation_step, outputs is a list of tensor scalars
        # we return just the average in this case (if we want)
        # return torch.stack(outputs).mean()

        val_loss_mean = 0
        val_acc_mean = 0
        for output in outputs:
            val_loss = output['val_loss']

            # reduce manually when using dp
            if self.trainer.use_dp or self.trainer.use_ddp2:
                val_loss = torch.mean(val_loss)
            val_loss_mean += val_loss

            # reduce manually when using dp
            val_acc = output['val_acc']
            if self.trainer.use_dp or self.trainer.use_ddp2:
                val_acc = torch.mean(val_acc)

            val_acc_mean += val_acc

        val_loss_mean /= len(outputs)
        val_acc_mean /= len(outputs)
        tqdm_dict = {'val_loss': val_loss_mean, 'val_acc': val_acc_mean}
        result = {'progress_bar': tqdm_dict, 'log': tqdm_dict, 'val_loss': val_loss_mean}
        return result

    # ---------------------
    # TRAINING SETUP
    # ---------------------
    def configure_optimizers(self):
        """
        return whatever optimizers we want here
        :return: list of optimizers
        """
        return [self.optimizer]

    @ptl.data_loader
    def train_dataloader(self):
        # logging.info('training data loader called')
        return DataLoader(self.train_dataset, batch_size=WAVENET_BATCH_SIZE, shuffle=True,
                          num_workers=NUM_WORKERS)

    @ptl.data_loader
    def val_dataloader(self):
        # logging.info('val data loader called')
        return DataLoader(self.eval_dataset, batch_size=WAVENET_BATCH_SIZE, shuffle=True, num_workers=NUM_WORKERS)

    @ptl.data_loader
    def test_dataloader(self):
        # logging.info('test data loader called')
        return DataLoader(self.test_dataset, batch_size=WAVENET_BATCH_SIZE, shuffle=True, num_workers=NUM_WORKERS)
```

**Comment by Vichoko** [comment]
Ran the same code in CPU and finished after 6 epochs with exit code 0.

`Epoch 6:  99%|█████████▉| 142/143 [13:59<00:04,  4.14s/batch, batch_nb=94, loss=0.654, train_loss=0.395, v_nb=1, val_acc=0.458, val_loss=0.754]
Epoch 6: 100%|██████████| 143/143 [14:01<00:00,  3.05s/batch, batch_nb=94, loss=0.654, train_loss=0.395, v_nb=1, val_acc=0.458, val_loss=0.778]
Epoch 6: 100%|██████████| 143/143 [14:01<00:00,  5.89s/batch, batch_nb=94, loss=0.654, train_loss=0.395, v_nb=1, val_acc=0.458, val_loss=0.778]

Process finished with exit code 0
`

**Comment by JSAustin** [comment]
Hello, I'm on a Windows 10 PC. In a previous version of lightning, I got no errors and it ran until it reached max epochs. Now, I reinstalled everything, including Conda, and with the same script, it terminates at random epochs as well with exit code 0. I don't think it's an early stopping thing, as my validation loss on a recent attempt kept decreasing continuously until it randomly stopped.

**Comment by awaelchli** [comment]
I also ran into this issue. According to the docs, the trainer has a default EarlyStopping defined. Setting this to None resolved the issue for me. However, I find it strange that early stopping takes effect even without defining validation_step/validation_end (in my case at least).

**Comment by Vichoko** [comment]
Nice finding. 
The corresponding docs are [here](https://williamfalcon.github.io/pytorch-lightning/Trainer/Training%20Loop/#early-stopping).

Any idea how's this default criteria defined?

**Comment by amrakm** [comment]
I'm still facing the same issue, nothing happens after training is done ( 100% epochs ) and nothing gets saved to disk
